internal/vpn/noise_credentials.go

```diff
@@ -0,0 +1,23 @@
+package vpn
+
+import (
+	"time"
+
+	"github.com/SkycoinProject/dmsg/cipher"
+)
+
+const (
+	// HSTimeout is a timeout for noise handshake.
+	HSTimeout = 5 * time.Second
+)
+
+// NoiseCredentials encapsulates sec and pub keys for noise usage.
+type NoiseCredentials struct {
+	PK cipher.PubKey
+	SK cipher.SecKey
+}
+
+// IsValid returns true only if PK and SK are valid.
+func (c NoiseCredentials) IsValid() bool {
+	return !c.PK.Null() && !c.SK.Null()
```

Comment:
I feel like we should only use the `SK` and generate the `PK` from the `SK`. This way, one has no chance of providing a key pair that are not associated with one another.

Reply:
Made this a bit different. this func is still the same, and SK may be nil if encryption is disabled. but in the `main` func of client and server I'm deriving PK out of SK if only SK is provided